### ISSUE TITLE
No module named matchpyramid

### ISSUE THREAD

**Issue by yifannie** [faneshion/MatchZoo#252]
Hi, I installed matchzoo in python 2.7.15 env, and when I tried to launch a matchpyramid model on my own dataset, it returned errors:
```
[ListGenerator] init done
Traceback (most recent call last):
  File "main.py", line 331, in <module>
    main(sys.argv)
  File "main.py", line 323, in main
    train(config)
  File "main.py", line 126, in train
    model = load_model(config)
  File "main.py", line 41, in load_model
    model = import_object(config['model']['model_py'], model_config)
  File "/u/xxx/projects/MatchZoo/matchzoo/utils/utility.py", line 30, in import_object
    return import_class(import_str)(*args, **kwargs)
  File "/u/xxx/projects/MatchZoo/matchzoo/utils/utility.py", line 20, in import_class
    __import__(mod_str)
ImportError: No module named matchpyramid
```
When I tried the command \__import\__("matchpyramid")  it showed the same error (No module named matchpyramid), it seemed that it could not find the relative import path... I tried to install the toolkit in py3, but still the same error, what should I do? Thanks.

**Comment by bwanglzu** [comment]
@yifannie Hi, thanks for your message! How can I reproduce the error? Can you provide a bit more context on that?

**Comment by yifannie** [comment]
@bwanglzu Hi bwanglzu, thanks a lot.
I created a clean new conda env called matchzoo under py2, then, I activated this conda env, then I cd into the matchzoo folder, and typed python setup.py install to install under this conda env.
As for the config file I used the example provided in the repo and modified my data path. I think the data loading process was OK, since it already prompted "list generator init done"...
my model config file is as follows:
``
{
  "net_name": "match_pyramid",
  "global":{
      "model_type": "PY",
      "weights_file": "/part/01/Tmp/nyf/exp/MatchPyramid/weights",
      "save_weights_iters": 10,
      "num_iters": 10,
      "display_interval": 10,
      "test_weights_iters": 10,
      "optimizer": "adam",
      "learning_rate": 0.0001
  },
  "inputs": {
    "share": {
        "text1_corpus": "/data/rali7/Tmp/nyf/MQ/MatchZoo/MQ2007/fold1.corpus.txt",
        "text2_corpus": "/data/rali7/Tmp/nyf/MQ/MatchZoo/MQ2007/fold1.corpus.txt",
        "use_dpool": true,
        "embed_size": 300,
        "vocab_size": 400002,
        "train_embed": false,
        "target_mode": "ranking",
        "text1_maxlen": 1000,
        "text2_maxlen": 1000
    },
    "train": {
        "input_type": "PairGenerator", 
        "phase": "TRAIN",
        "use_iter": false,
        "query_per_iter": 50,
        "batch_per_iter": 5,
        "batch_size": 100,
        "relation_file": "/data/rali7/Tmp/nyf/MQ/MatchZoo/MQ2007/fold1.rel"
    },
    "valid": {
        "input_type": "ListGenerator", 
        "phase": "EVAL",
        "batch_list": 10,
        "relation_file": "/data/rali7/Tmp/nyf/MQ/MatchZoo/MQ2007/S4.rel"
    },
    "test": {
        "input_type": "ListGenerator", 
        "phase": "EVAL",
        "batch_list": 10,
        "relation_file": "/data/rali7/Tmp/nyf/MQ/MatchZoo/MQ2007/S5.rel"
    },
    "predict": {
        "input_type": "ListGenerator", 
        "phase": "PREDICT",
        "batch_list": 100,
        "relation_file": "/data/rali7/Tmp/nyf/MQ/MatchZoo/MQ2007/S5.rel"
    }
  },
  "outputs": {
    "predict": {
      "save_format": "TREC",
      "save_path": "predict.test.matchpyramid_ranking.txt"
    }
  },
  "model": {
    "model_path": "/part/01/Tmp/nyf/exp/MatchPyramid/",
    "model_py": "matchpyramid.MatchPyramid",
    "setting": {
        "kernel_count": 32, 
        "kernel_size": [3, 3], 
        "dpool_size": [3, 10],
        "dropout_rate": 0
    }
  },
  "losses": [ 
    {
       "object_name": "rank_hinge_loss" ,
       "object_params": {
            "margin": 1.0
       }
    }
  ],
  "metrics": [ "ndcg@3", "ndcg@5", "map" ]
}
``
I didn't touch the "model_py" path and their name, I just kept them as in the example.

Thanks.

**Comment by bwanglzu** [comment]
@yifannie can you change the `model_path` to "matchzoo/models/"?

**Comment by yifannie** [comment]
@bwanglzu Hi bwanglzu, thanks a lot, I changed the model_path to "matchzoo/models/"
```
"model": {
    "model_path": "matchzoo/models/",
    "model_py": "matchpyramid.MatchPyramid",
    "setting": {
        "kernel_count": 32, 
        "kernel_size": [3, 3], 
        "dpool_size": [3, 10],
        "dropout_rate": 0
    }
```
```
But it still reported error:
List Instance Count: 339
[ListGenerator] init done
Traceback (most recent call last):
  File "main.py", line 331, in <module>
    main(sys.argv)
  File "main.py", line 323, in main
    train(config)
  File "main.py", line 126, in train
    model = load_model(config)
  File "main.py", line 41, in load_model
    model = import_object(config['model']['model_py'], model_config)
  File "/u/nieyifan/projects/MatchZoo/matchzoo/utils/utility.py", line 30, in import_object
    return import_class(import_str)(*args, **kwargs)
  File "/u/nieyifan/projects/MatchZoo/matchzoo/utils/utility.py", line 20, in import_class
    __import__(mod_str)
ImportError: No module named matchpyramid

```
I think the installation of matchzoo was successful because when I just type import matchzoo, it reported no error...

**Comment by bwanglzu** [comment]
Can you try ./matchzoo/models/ , or use absolute path (path-to-/matchzoo/models/) ?

**Comment by yifannie** [comment]
@bwanglzu Hi, thanks a lot for your help.
I tried ./matchzoo/models/. It didn't work, but the absolute path worked. 
The problem is resolved.
Thanks a lot.

For your info, my OS Is Debian: Linux  3.16.0-6-amd64 #1 SMP Debian 3.16.56-1+deb8u1 (2018-05-08) x86_64 GNU/Linux
python env is:
Python 2.7.15 |Anaconda, Inc.| (default, May  1 2018, 23:32:55)
[GCC 7.2.0] on linux2

and I created a clean conda env to install this package.
